### PR TITLE
[semver:major] update apply tf to new flow

### PR DESCRIPTION
**Pull request by maxeshleman**
#### Details
- update apply tf to use s3 backend

### CODE REVIEW

File: src/jobs/apply-tf.yml [hunk 1]
```diff
@@ -2,42 +2,45 @@ working_directory: ~/repo/tf
...
-  workspace:
-    description: "Workspace to run Terraform apply against"
-    default: "dev"
+  environment:
+    description: "Terraform environment name"
+    type: string
+  service:
+    description: "The name of the service being deployed"
+    type: string
+  bucket:
```
Comment (chronoslynx) [inline src/jobs/apply-tf.yml:11]: Could we infer these from the environment's name instead?

File: src/scripts/apply-tf.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  echo "Must provide AWS_PROFILE environment variable"
+  exit 1
+fi
+
+VERSION=$(cat ../VERSION)
+if [ -n "${USE_VERSION_SHA:-}" ]; then
+  VERSION="$VERSION-${CIRCLE_SHA1:-}"
+fi
+
+aws ssm get-parameter --name "/cast/${ENV_NAME}/vars/${SERVICE}" --query 'Parameter.Value' --output text > vars.auto.tfvars
```
Comment (chronoslynx) [inline src/scripts/apply-tf.sh:34]: Have you documented what these params should look like anywhere?

File: src/jobs/ginkgo.yml [hunk 3]
```diff
@@ -14,9 +14,9 @@ parameters:
     type: string
     description: "a comma separated list of packages to skip"
     default: ""
-  goimports:
+  linter:
     type: boolean
-    description: "run goimports validation"
+    description: "run golangci lint"
```
[the remark this answers is not in the document]
Comment (maxeshleman) [reply src/jobs/ginkgo.yml:19]: YES I didn't mean to commit it yet :(

File: src/jobs/apply-tf.yml (hunk 1 above)
Comment (maxeshleman) [reply src/jobs/apply-tf.yml]: I am now grabbing them from ssm params

File: src/jobs/plan-tf.yml [hunk 4]
```diff
@@ -14,27 +16,32 @@ parameters:
...
+        ENV_NAME: <<parameters.environment>>
+        AWS_PROFILE: terraform
+      command: <<include(scripts/init-tf.sh)>>
   - run:
+      name: set terraform variables
       command: |
         VERSION=$(cat ../VERSION)
-        echo "app-version = \"$VERSION<<# parameters.use_version_sha >>-$CIRCLE_SHA1<</ parameters.use_version_sha >>\"" > version.auto.tfvars
         echo "app_version = \"$VERSION<<# parameters.use_version_sha >>-$CIRCLE_SHA1<</ parameters.use_version_sha >>\"" >> version.auto.tfvars
+        aws ssm get-parameter --name "/cast/${ENV_NAME}/vars/${SERVICE}" --query 'Parameter.Value' --output text > vars.auto.tfvars
```
Comment (chronoslynx) [inline src/jobs/plan-tf.yml:43]: I love this approach

File: src/scripts/init-tf.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [ -z "${ENV_NAME:-}" ]; then
+  echo "Must provide ENV_NAME environment variable"
+  exit 1
+fi
+
+STATE_BUCKET=$(aws ssm get-parameter --name "/cast/${ENV_NAME}/state_bucket" --query 'Parameter.Value' --output text)
```
Comment (chronoslynx) [inline src/scripts/init-tf.sh:9]: I also love this

**Review by chronoslynx — APPROVED** [review]

**Pull request merged**